user-registration-app/src/app/views/user-profile-view/user-profile-view.component.ts

```diff
@@ -1,178 +1,135 @@
-    }
-    return user.photoURL;
-  }
-
-  editEmail() {
-    this.emailEditToggled = true;
-  }
-
-  editPassword() {
-    this.passwordEditToggled = true;
-  }
-
-  editName() {
-    this.nameEditToggled = true;
+  public getUserPhotoUrl(user: app.User | null) {
+    return user.photoURL || UserProfileViewComponent.DEFAULT_PROFILE_URL;
   }
 
   submitNewName(value: string) {
-    this.progressService.start();
+    this.progressService.ref().start();
     const observable = this.authService.currentUser
       .pipe(
-        switchMap((user) => {
-          return user.updateProfile({ displayName: value, photoURL: user.photoURL });
-        }),
+        switchMap(user => user.updateProfile({ displayName: value, photoURL: user.photoURL })),
         finalize(() => {
-          this.progressService.complete();
-          this.nameEditToggled = false;
+          this.progressService.ref().complete();
         }),
-      ).subscribe(() => {
-        this.alertsService.info('Successfully changed display name');
-        observable.unsubscribe();
-      }, (error) => {
-        console.error(error);
-        this.alertsService.danger('Something went wrong. Try again');
-        observable.unsubscribe();
-        return undefined;
-      });
+      ).subscribe(
+        () => {
+          this.toastrService.info('Successfully changed display name');
+          observable.unsubscribe();
+        },
+        (error) => {
+          console.error(error);
+          this.toastrService.error('Something went wrong. Try again');
+          observable.unsubscribe();
+        },
+      );
   }
 
   submitNewEmail(value: string) {
-    this.progressService.start();
+    this.progressService.ref().start();
     const observable = this.authService.currentUser
       .pipe(
-        switchMap((user) => {
-          return user.updateEmail(value);
-        }),
-        finalize(() => {
-          this.progressService.complete();
-          this.nameEditToggled = false;
-        }),
-      ).subscribe(() => {
-        this.alertsService.info('Successfully changed display name');
-        observable.unsubscribe();
-      }, (error) => {
-        console.error(error);
-        this.alertsService.danger(error.message);
-        observable.unsubscribe();
-        return undefined;
-      });
+        switchMap(user => user.updateEmail(value)),
+        finalize(() => this.progressService.ref().complete()),
+      ).subscribe(
+        () => {
+          this.toastrService.info('Successfully changed display name');
+          observable.unsubscribe();
+        },
+        (error) => {
+          console.error(error);
+          this.toastrService.error(error.message);
+          observable.unsubscribe();
+        },
+      );
   }
 
-  submitNewPassword(value1: string, value2: string) {
-    if (value1.length === 0 || value2.length === 0) {
-      this.alertsService.danger('Password not entered');
+  submitNewPassword(newPassword: string, confirmedPassword: string) {
```

Comment:
Stan, are you trying to say that value1 and value2 are _not_ good variable names?? lmao

Reply:
Let's just say my standards go a little higher than what I found in the code base :candy: